### ISSUE TITLE
Roster incoherence when removing a member that is also in another roster

### ISSUE THREAD

**Issue by jlemangarin** [jsxc/jsxc#793]
### Expected behavior
The contact is not removed from JSXC
### Actual behavior
The contact is removed from JSXC until next roster fetch
### Steps to reproduce the behavior
1.Create two shared rosters with you and another member
2.Delete the other member in the first roster (the member is still with you in the other roster)
3.

### Environment
- **JSXC version:** 4.0.0-b3
- **Host system and version:** CentOS 7
- **Browser vendor and version:** Chrome 75.0.3770.80
- **Any browser plugins enabled?** ARC & ITHit but nothing that can alter this behavior
- **XMPP server vendor and version:** Ejabberd 19.05
- **Is your XMPP server working with other clients as expected?** Yes (mobile app)

#### Javascript
//

#### JSXC
<item xmlns="jabber:iq:roster" subscription="both" jid="smassonnat@gofast-comm-internal.ceo-vision.com">
    <group>**4_Group**</group>
    <group>**5_Orga**</group>
</item>
.......
<iq xmlns="jabber:client" to="jleman@gofast-comm-internal.ceo-vision.com/jsxc-d4c069ff" from="jleman@gofast-comm-internal.ceo-vision.com" type="set" id="push14351270947443124645">
    <query xmlns="jabber:iq:roster">
        <item subscription="remove" jid="smassonnat@gofast-comm-internal.ceo-vision.com">
            <group>**5_Orga**</group>
        </item>
    </query>
</iq>

#### XMPP
//

#### Host
//

#### Informations
The XMPP server also send the name of the shared rosters so I think JSXC should save on what roster the contact is and delete the contact only if the contact has been removed from all rosters

I recorded a video for a better understanding of the issue : https://gofast.ceo-vision.com/node/60929

Also, I can work on the issue and make a pull request. :)

**Comment by sualko** [comment]
The server is sending a roster update with subscription "remove", according to https://tools.ietf.org/html/rfc6121#section-2.1.6 the item should be removed. Therefore it sounds more like an ejabberd issue.

**Comment by jlemangarin** [comment]
Yes you're right, this is the concerned part I think : 

`Sometime later, the user might want to remove the item from the
   original group.

   C: <iq from='juliet@example.com/balcony'
          id='lf72v157'
          type='set'>
        <query xmlns='jabber:iq:roster'>
          <item jid='romeo@example.net'
                name='Romeo'>
            <group>Lovers</group>
          </item>
        </query>
      </iq>`

**Comment by Neustradamus** [comment]
Any news about it?

**Comment by jlemangarin** [comment]
Nope, for myself I had to give up the idea of roster and base my chat on subscribed contacts.

I think you should up the ejabberd issue instead of this one because it's ejabberd that is not sending the proper request to the client..
